### PR TITLE
br: fix checksum incomplete when context is done

### LINKED ISSUE #44472: br: when full restore to multi rocksdb cluster, checksum will be mismatch sometimes
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. full restore to multi rocksdb cluster

### 3. What did you see instead (Required)
1. checksum mismatch
2. but later checksum can be ok
### 4. What is your TiDB version? (Required)
sh-5.1# /br --version
Release Version: v7.1.0
Git Commit Hash: fd2314159d1a81c8b13e5d61e53dcd527cb35494
Git Branch: heads/refs/tags/v7.1.0
Go Version: go1.20.3
UTC Build Time: 2023-05-24 02:08:51
Race Enabled: false

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: close #44472 

Problem Summary:
When BR is failed by one checksum task, the other checksum tasks' result would be incomplete.

### What is changed and how it works?
check the context status before returns response.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by tonyxuqqi — APPROVED** [review]

File: br/pkg/checksum/executor.go [hunk 1]
```diff
@@ -365,5 +365,18 @@ func (exec *Executor) Execute(
...
 		updateFn()
 	}
-	return checksumResp, nil
+	return checksumResp, checkContextDone(ctx)
+}
+
+// The coprocessor won't return the error if the context is done,
+// so sometimes BR would get the incomplete result.
+// checkContextDone makes sure the result is not affected by CONTEXT DONE.
+func checkContextDone(ctx context.Context) error {
```
Comment (YuJuncen) [inline br/pkg/checksum/executor.go:374]: In fact you can just call `ctx.Err() != nil`.
```go
// If Done is not yet closed, Err returns nil.
	// If Done is closed, Err returns a non-nil error explaining why:
	// Canceled if the context was canceled
	// or DeadlineExceeded if the context's deadline passed.
	// After Err returns a non-nil error, successive calls to Err return the same error.
	Err() [error](https://pkg.go.dev/builtin#error)
```

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/checksum/executor.go (hunk 1 above)
Comment (Leavrth) [reply br/pkg/checksum/executor.go:374]: done

**Review by 3pointer — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**